### ISSUE TITLE
Improving Keras Sequential Neural Network Reproducability

### ISSUE THREAD

**Issue by MakiaP** [googlecolab/colabtools#3031]
I am a Data Science doctoral student and am having trouble reproducing a Keras Sequential Neural Network which takes constant Tensors as input so that my JupyterLab Tensorflow environment has the same results as Google Colab Pro. More specifically even though I use the same Python code and same input dataset and set random seeds the accuracy metrics differ by 10 percent. Also the versions of Keras and Tensorflow in both environments is set to 2.8. In addition, the Google Colab Pro Keras Sequential Neural Network outputs " Training on 3000 samples" while the JupyterLab Keras lacks this output despite having the same version, 2.8. In order to be able to publish results, the accuracy needs to be reproducible.

**Describe the solution you'd like**
This accuracy discrepancy appears to be related to randomization. If we could have further methods to specify how the inputs are randomized and shuffled for tensor inputs that would be great. This capability appears to already be available for Keras Sequences and Keras DataSets but appears to be lacking for Tensor inputs.

**Describe alternatives you've considered**
I have tried converting the tensor inputs to Keras Sequences and Keras DataSets while specifying a random seed and am able to reproduce accuracy results on the same code. However, the overall accuracy is 10 percent lower than Tensor inputs on Google Colab Pro.

**Additional context**
# The Keras code is as follows 
# Convert all Pandas Dataframes to Tensors
X_train = tf.constant(total_x, dtype = tf.float32)
Y_train = tf.constant(total_y_t, 'float32')
X_test = tf.constant(total_x_test, dtype = tf.float32)
Y_test = tf.constant(total_y_test_t, 'float32')

# Set random seed
tf.random.set_seed(10)

# Create a model (same as above)
model = tf.keras.Sequential([
  tf.keras.layers.Dense(200, activation="swish"), 
  tf.keras.layers.Dropout(0.05, seed= 10),
  tf.keras.layers.Dense(200, activation="swish"), 
  tf.keras.layers.Dropout(0.05, seed= 10),
  tf.keras.layers.Dense(200, activation="swish"), 
  tf.keras.layers.Dense(1) # 1 unit (important for output layer)
])

# Compile model (same as above)
model.compile(loss=tf.keras.losses.mae,
                          optimizer=tf.keras.optimizers.Adam(),
                          metrics=['mae', 'accuracy'])

# Fit model (same as above)
model.fit(X_train, Y_train, batch_size=50, steps_per_epoch=3000, epochs=10)

# Predict Labels
Y_predict = model.predict(X_test, steps=3000)

# Compute summary
model.evaluate(X_test, Y_test, steps = 16000, batch_size = 50)
